packages/cli/src/commands/trace.ts

```diff
@@ -0,0 +1,105 @@
+import type { CommandModule } from "yargs";
+import { ethers } from "ethers";
+import { loadConfig } from "@latticexyz/config/node";
+import { StoreConfig } from "@latticexyz/store";
+import { cast, getSrcDirectory } from "@latticexyz/common/foundry";
+import { resolveWorldConfig, WorldConfig } from "@latticexyz/world";
+import { TableId } from "@latticexyz/utils";
+import { IBaseWorld } from "@latticexyz/world/types/ethers-contracts/IBaseWorld";
+import { getChainId, getExistingContracts } from "../utils";
+
+import IBaseWorldData from "@latticexyz/world/abi/IBaseWorld.sol/IBaseWorld.json" assert { type: "json" };
+import { existsSync, readFileSync } from "fs";
+import { MUDError } from "@latticexyz/common/errors";
+
+const systemsTableId = new TableId("", "Systems");
+
+type Options = {
+  tx: string;
+  worldAddress?: string;
+  configPath?: string;
+  srcDir?: string;
+  rpc?: string;
+};
+
+const commandModule: CommandModule<Options, Options> = {
+  command: "trace",
+
+  describe: "Display the trace of a transaction",
+
+  builder(yargs) {
+    return yargs.options({
+      tx: { type: "string", required: true, description: "Transaction hash to replay" },
+      worldAddress: {
+        type: "string",
+        description: "World contract address. Defaults to the value from worlds.json, based on rpc port",
+      },
+      configPath: { type: "string", desc: "Path to the config file" },
+      srcDir: { type: "string", desc: "Source directory. Defaults to foundry src directory." },
+      rpc: { type: "string", description: "json rpc endpoint, defaults to http://localhost:8545" },
```

Comment:
we should make this explicitly default to `localhost:8545` with the `default` option here. (Then we also don't need to name the default in the description because it will be displayed by default).

Reply:
actually we should probably default to the rpc of the foundry config, and just fall back to localhost if it doesn't exist